### DISCUSSION TITLE
[Admin UI] Edit style of login page

### DISCUSSION THREAD

**Discussion by farnots** [medusajs/medusa#5942] [Q&A] [answered]
Hello 👋🏻 

I saw in the medusa documentation https://docs.medusajs.com/admin/routes that all routes are prefixed by `/a/` 

I want to rework the `/login` page https://github.com/medusajs/medusa/blob/develop/packages/admin-ui/ui/src/pages/login.tsx to add custom logo / text / ...

How can I archive that?

Thanks a lot

**Comment by WalobwaD** [comment]
Hey
Any help you got?
I'm also trying to achieve this but I can't find any solution to it.

**Comment by saulgudmon** [comment]
From what I can tell, the admin extension system doesn't give you the tools to white label the login page. If you just want to add stuff you can [create an admin widget](https://docs.medusajs.com/admin/widgets) with the `zone: "login.before"` or `zone: "login.after"` config. 

example:
```tsx
// path: medusa-backend/src/admin/widgets/

import type { WidgetConfig } from "@medusajs/admin"

const LoginCustomizationsWidget = () => {
    return (
            <p className="text-center">custom stuff</p>
    )
}

export const config: WidgetConfig = {
    zone: "login.before",
}

export default LoginCustomizationsWidget
```
![image](https://github.com/medusajs/medusa/assets/33010393/0cd06318-a320-4a58-b051-2cdbad8e94fd)

[It seems like if you wanted to do anything more advanced than this to the login page it would be quite difficult](https://github.com/medusajs/medusa/discussions/5898#discussioncomment-7906997). From what I can tell, you would need to fork [both admin and admin-ui packages](https://github.com/medusajs/medusa/blob/v1.19.0/packages/) and then resolve to your fork of admin instead of the core package in `medusa-config.js` (in the plugins array), then you would need to modify your fork of the admin package to include your fork of the admin-ui package instead of the core package. The medusa team recommend against doing this. You would of course be more likely to have to deal with conflicts and other issues when updating the core packages and merging updated admin packages with your forks. Also, I have not done this before so it is likely I could be missing something in my directions, but if you decide to try it please let me know how it goes. Good luck

**Reply by WalobwaD** [reply]
Hey. Thanks but I already tried the admin widgets and it's not exactly what I wanted.
I found another solution that works quite well. I directly modified the @medusajs/admin-ui package, it has the whole frontend code so I modified every part of the admin ui I needed to. Then used the patch package to make a patch file of the package. Works perfect on my end. But I'm not yet sure if it's safe doing it.

Here's the output of what I tried.

![Screenshot 2023-12-27 225350](https://github.com/medusajs/medusa/assets/83467845/4b574886-6397-4e5c-8200-7a98da7f7d82)

**Reply by saulgudmon** [reply]
Is [this](https://www.npmjs.com/package/patch-package) the patch package you are referring to? Interesting. I always thought it was bad practice to modify anything in the node_modules directory, but this is probably a better path to take as it just occurred to me that with the method I described you would probably have to fork any medusa plugins you were using and modify them to act on your admin forks as well

**Reply by WalobwaD** [reply]
Yeah. That's the one. Pretty simple to use.

**Reply by WalobwaD** [reply]
But I'm open to any advice. If it's really an ok thing to do or I should default to another method.

**Reply by saulgudmon** [reply]
Thank you. I had never heard of patch-package before. It seems like the perfect workaround for modifying the templates in the admin-ui. I guess you will know for sure after you update medusa core in your project a couple times.

**Reply by alightinastorm** [reply]
If you really feel like removing the medusa logo, your custom widget could remove the medusa logo dom node on mount..

**Reply by saulgudmon** [reply, marked as answer]
> If you really feel like removing the medusa logo, your custom widget could remove the medusa logo dom node on mount..

idk why I didn't think of that.. 🥴

```tsx
import { useEffect } from 'react';
import type { WidgetConfig } from "@medusajs/admin"

const LoginCustomizationsWidget = () => {
    useEffect(() => {
        // remove logo
        document.querySelectorAll('.mb-large').forEach((DOMnode) => DOMnode.remove())
    }, [])
    
    return (            
            <p className="text-center">custom stuff</p>
    )
}

export const config: WidgetConfig = {
    zone: "login.before",
}

export default LoginCustomizationsWidget
```

**Reply by farnots** [reply]
> If you really feel like removing the medusa logo, your custom widget could remove the medusa logo dom node on mount..


Yes that was our solution too but that mean looking in the DOM to find the Medusa Logo and replace it with our own. Not really clean code

**Reply by farnots** [reply]
Thanks @WalobwaD for your suggestion. How `patch-package` works ? At every medusa update it could be broken non ?

**Reply by alightinastorm** [reply]
> > If you really feel like removing the medusa logo, your custom widget could remove the medusa logo dom node on mount..
> 
> 
> 
> 
> 
> Yes that was our solution too but that mean looking in the DOM to find the Medusa Logo and replace it with our own. Not really clean code

of course, the best is to just let the medusa logo stay there
